keystorage/src/main/java/tech/pegasys/web3signer/keystorage/awskms/AwsKeyManagerService.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.keystorage.awskms;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static tech.pegasys.web3signer.common.config.AwsAuthenticationMode.ENVIRONMENT;
+
+import tech.pegasys.web3signer.common.config.AwsAuthenticationMode;
+import tech.pegasys.web3signer.common.config.AwsCredentials;
+
+import java.net.URI;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.kms.KmsClientBuilder;
+import software.amazon.awssdk.services.kms.model.GetPublicKeyRequest;
+import software.amazon.awssdk.services.kms.model.GetPublicKeyResponse;
+import software.amazon.awssdk.services.kms.model.KeySpec;
+import software.amazon.awssdk.services.kms.model.MessageType;
+import software.amazon.awssdk.services.kms.model.SignRequest;
+import software.amazon.awssdk.services.kms.model.SignResponse;
+import software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec;
+
+/**
+ * Wraps AWS KMS Client and expose public key and sign operations. Since KmsClient is thread safe,
+ * same instance of this class can be used to talk to same host/credentials.
+ */
+public class AwsKeyManagerService implements AutoCloseable {
+  private final AwsAuthenticationMode authMode;
+  private final AwsCredentials awsCredentials;
+  private final String region;
+
+  private KmsClient kmsClient;
+
+  private Optional<URI> endpointOverride;
+
+  public AwsKeyManagerService(
```

Comment:
nit: we have classes with the full name and some with AwsKMS, think it would be better to be consistent one way or the other for the sake of searchability

Reply:
Or maybe you intended for the difference in name to reflect two different levels of abstraction?

Reply:
I wanted this class to be wrapper around AWS' KmsClient. For secrets manager, we had used name `AwsSecretsManager` ... `secrets manager` is the service we are using. On the same note, I used AwsKeyManagerService, `key manager service` is the service we are using. However, I agree with your initial observation, this can be easily rename to `AwsKMS` as in the signing module, we are using `KMS`.